### PR TITLE
docs: add python constraint docs

### LINKED ISSUE #1674: Add missing Python constraint docs
#### Reason/Context
In the other languages, we have explicit documentation explaining the constraint rules such as for TypeScript: https://modelina.org/docs/constraints/typescript

However, we are missing one for Python.

TODO:
- Go over the constraint logic for Python for constants: https://github.com/asyncapi/modelina/blob/master/src/generators/python/constrainer/ConstantConstrainer.ts
- Go over the constraint logic for Python for enum value and key constraints: https://github.com/asyncapi/modelina/blob/master/src/generators/python/constrainer/EnumConstrainer.ts
- Go over the constraint logic for Python for model naming constraints: https://github.com/asyncapi/modelina/blob/master/src/generators/python/constrainer/ModelNameConstrainer.ts
- Go over the constraint logic for Python for property key constraints: https://github.com/asyncapi/modelina/blob/master/src/generators/python/constrainer/PropertyKeyConstrainer.ts

You could potentially copy-paste the TypeScript one: https://github.com/asyncapi/modelina/blob/master/docs/constraints/TypeScript.md and adapt it to Python

### PR DESCRIPTION
**Pull request by devilkiller-ag**
**Description**
This PR adds documentation for explaining the constraint rules in python

**Related issue(s)**
Fixes #1674

### CODE REVIEW

File: docs/constraints/Python.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Python Constraints
+
+These are the Python specific constraints applied to the MetaModel before reaching the presets. [Read here to get a more general idea on the overall process](../input-processing.md) of converting a MetaModel to a ConstrainedMetaModel.
+
+## Model Naming
+These are the constraints that is applied to model naming. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](../constraints.md#Customization).
```
Comment (jonaslagoni) [inline docs/constraints/Python.md:6]:
```suggestion
These are the constraints that is applied to model naming. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](./README.md#Customization).
```

File: docs/constraints/Python.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+|NO_SPECIAL_CHAR|No special characters| Special characters are replaced by their name, for example `!` is replaced with `exclamation`. For python `_` is an exception to this rule. |
+|NO_NUMBER_START_CHAR|No numbers as starting characters|Default behavior is pre pending `number_` in front of the first character|
+|NO_EMPTY_VALUE|No empty values|Default behavior is to use `empty` as name. |
+|NO_RESERVED_KEYWORDS|No reserved keywords|Python has a list of reserved keywords ([see the full list here](../../src/generators/python/Constants.ts))|
+|NAMING_FORMATTER|Must be formatted equally|Property name is formatted using camel case|
+|NO_DUPLICATE_PROPERTIES|No duplicate properties|If any of the above constraints changes the property name, we must make sure that no duplicates exist within the same object. If any is encountered `reserved_` is pre-pended. This is done recursively until no duplicates are found.| 
+
+
+## Enum key constraints
+These are the constraints that is applied to enum keys. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](../constraints.md#Customization).
```
Comment (jonaslagoni) [inline docs/constraints/Python.md:29]:
```suggestion
These are the constraints that is applied to enum keys. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](./README.md#Customization).
```

File: docs/constraints/Python.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+|Rule key|Rule|Resolution|
+|---|---|---|
+|NO_SPECIAL_CHAR|No special characters| Special characters are replaced by their name, for example `!` is replaced with `exclamation`. For python `_` is an exception to this rule. |
+|NO_NUMBER_START_CHAR|No numbers as starting characters|Default behavior is pre pending `number_` in front of the first character|
+|NO_EMPTY_VALUE|No empty values|Default behavior is to use `empty` as name. |
+|NO_RESERVED_KEYWORDS|No reserved keywords|Python has a list of reserved keywords ([see the full list here](../../src/generators/python/Constants.ts))|
+|NAMING_FORMATTER|Must be formatted equally|Model name is formatted using pascal case|
+
+## Property naming
+These are the constraints that is applied to object properties and the naming of them. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](../constraints.md#Customization).
```
Comment (jonaslagoni) [inline docs/constraints/Python.md:17]:
```suggestion
These are the constraints that is applied to object properties and the naming of them. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](./README.md#Customization).
```

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍 Just had to change a few links 🙂

**Comment by jonaslagoni** [comment]
@all-contributors please add @devilkiller-ag for docs

**Comment by jonaslagoni** [comment]
@all-contributors please add @devilkiller-ag for doc, maintenance

**Comment by devilkiller-ag** [comment]
Thanks @jonaslagoni ✨

**Pull request merged**